employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/pages/rotation/ShiftTemplateEditForm.java

```diff
@@ -0,0 +1,242 @@
+        shiftTemplateGridObject.getElement().classList.add(SELECTED_CLASS);
+        final ShiftTemplateView template = shiftTemplateGridObject.getShiftTemplateModel().getShiftTemplateView();
+        setup(template);
+        showFor(shiftTemplateGridObject);
+    }
+
+    public void createNewShiftTemplate() {
+        setup(new ShiftTemplateView(tenantStore.getCurrentTenantId(), new ShiftTemplate(tenantStore.getCurrentTenantId(), new Spot(), 0, LocalTime.MIDNIGHT, 1, LocalTime.MIDNIGHT)));
+        spotSelect.setEnabled(true);
+        deleteButton.remove();
+        setTitle(translationService.format(I18nKeys.ShiftRosterToolbar_createShift));
+        show();
+    }
+
+    private void setup(ShiftTemplateView shiftTemplateView) {
+        employeeSelect.clear();
+        employeeSelect.addItem(translationService.format(I18nKeys.Shift_unassigned), "-1");
+
+        SpotRestServiceBuilder.getSpotList(tenantStore.getCurrentTenantId(), FailureShownRestCallback.onSuccess(spots -> {
+            spots.forEach(s -> this.spotSelect.addItem(s.getName(), s.getId().toString()));
+            Optional<Spot> mySpot = spots.stream().filter(s -> s.getId().equals(shiftTemplateView.getSpotId())).findAny();
+            if (mySpot.isPresent()) {
+                spotSelect.setSelectedIndex(spots.indexOf(mySpot.get()));
+            } else {
+                spotSelect.setSelectedIndex(0);
+            }
+        }));
+
+        EmployeeRestServiceBuilder.getEmployeeList(tenantStore.getCurrentTenantId(), FailureShownRestCallback.onSuccess(employees -> {
+            employees.forEach(e -> employeeSelect.addItem(e.getName(), e.getId().toString()));
+            Optional<Employee> myEmployee = employees.stream().filter(s -> s.getId().equals(shiftTemplateView.getRotationEmployeeId())).findAny();
+            if (myEmployee.isPresent()) {
+                employeeSelect.setSelectedIndex(employees.indexOf(myEmployee.get()) + 1);
+            } else {
+                employeeSelect.setSelectedIndex(0);
+            }
+        }));
+
+        long startDayOffset = shiftTemplateView.getDurationBetweenReferenceAndStart().toDays();
+        from.setDayOffset((int) startDayOffset);
+        from.setTime(LocalTime.MIDNIGHT.plusSeconds(shiftTemplateView.getDurationBetweenReferenceAndStart().minusDays(startDayOffset).getSeconds()));
+        long endDayOffset = shiftTemplateView.getDurationBetweenReferenceAndStart().plus(shiftTemplateView.getShiftTemplateDuration()).toDays();
+        to.setDayOffset((int) endDayOffset);
+        to.setTime(LocalTime.MIDNIGHT.plusSeconds(shiftTemplateView.getDurationBetweenReferenceAndStart().plus(shiftTemplateView.getShiftTemplateDuration()).minusDays(endDayOffset).getSeconds()));
+    }
+
+    @Override
+    protected void onClose() {
+        if (shiftTemplateGridObject != null) {
+            shiftTemplateGridObject.getElement().classList.remove(SELECTED_CLASS);
```

Comment:
I think that the code would be more concise and easier to read, test and maintain if the class manipulation was encapsulated in the `ShiftTemplateGridObject`. This could then be expressed as
```java
shiftTemplateGridObject.unselect();
// or
shiftTemplateGridObject.setSelected(false);
```

Reply:
Will do

Reply:
Done